### PR TITLE
Harrison/sql query

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: libs/langchain/langchain/chains/sql_database/query.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    formatted_prompt = {
+        "input": lambda x: x["question"] + "\nSQLQuery: ",
+        "dialect": lambda _: db.dialect,
+        "top_k": lambda _: k,
+        "table_info": lambda x: db.get_table_info(
+            table_names=x.get("table_names_to_use")
+        ),
+    } | prompt_to_use
+    return (
+        formatted_prompt | llm.bind(stop=["\nSQLResult:"]) | NoOpOutputParser() | _strip
```
Comment (baskaryan) [inline libs/langchain/langchain/chains/sql_database/query.py:37]: think you can just str.strip instead of _strip

Comment (hinthornw) [reply libs/langchain/langchain/chains/sql_database/query.py:37]: ooc: do lambdas work here too?

File: libs/langchain/langchain/chains/sql_database/query.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    else:
+        prompt_to_use = PROMPT
+    formatted_prompt = {
+        "input": lambda x: x["question"] + "\nSQLQuery: ",
+        "dialect": lambda _: db.dialect,
+        "top_k": lambda _: k,
+        "table_info": lambda x: db.get_table_info(
+            table_names=x.get("table_names_to_use")
+        ),
+    } | prompt_to_use
```
Comment (hinthornw) [inline libs/langchain/langchain/chains/sql_database/query.py:35]: Oh this actually seems more confusing to me right now

File: libs/langchain/langchain/chains/sql_database/query.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    return text.strip()
+
+
+def create_sql_query_chain(
+    llm: BaseLanguageModel,
+    db: SQLDatabase,
+    prompt: Optional[BasePromptTemplate] = None,
+    k: int = 5,
+) -> RunnableSequence:
+    if prompt is not None:
```
Comment (baskaryan) [inline libs/langchain/langchain/chains/sql_database/query.py:22]: if part of goal is to show how succinctly we can create this chain:
```python
prompt = prompt or SQL_PROMPTS.get(db.dialect, PROMPT)
```

File: libs/langchain/langchain/chains/sql_database/query.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        prompt_to_use = prompt
+    elif db.dialect in SQL_PROMPTS:
+        prompt_to_use = SQL_PROMPTS[db.dialect]
+    else:
+        prompt_to_use = PROMPT
+    formatted_prompt = {
```
Comment (baskaryan) [inline libs/langchain/langchain/chains/sql_database/query.py]:
```python
input_to_prompt_vars = {...}
return input_to_prompt_vars | prompt_to_use | llm...
```
more readable imo

**Pull request merged**